exporters/trace/jaeger/agent.go

```diff
@@ -124,6 +132,46 @@ func (a *agentClientUDP) EmitBatch(ctx context.Context, batch *gen.Batch) error
 	return err
 }
 
+// calcSizeOfSerializedThrift calculate the serialized thrift packet size
+func (a *agentClientUDP) calcSizeOfSerializedThrift(ctx context.Context, thriftStruct thrift.TStruct) int {
+	a.thriftBuffer.Reset()
+	_ = thriftStruct.Write(ctx, a.thriftProtocol)
+	return a.thriftBuffer.Len()
+}
+
+// splitAndFlush split large batch into small ones to fit the udp packet limit
+func (a *agentClientUDP) splitAndFlush(ctx context.Context, batch *gen.Batch) error {
+	var err error
+	processSize := a.calcSizeOfSerializedThrift(ctx, batch.Process)
+	totalSize := processSize
+	var spans []*gen.Span
+	for _, span := range batch.Spans {
+		spanSize := a.calcSizeOfSerializedThrift(ctx, span)
+		if spanSize+processSize >= a.maxPacketSize {
+			// drop the span that exceeds the limit
+			continue
+		}
+		if totalSize+spanSize >= a.maxPacketSize {
+			err = a.Flush(ctx, &gen.Batch{
```

Comment:
Does this work? `a.Flush` calls `EmitBatch`, `EmitBatch` calls this `splitAndFlush`. This looks like an infinite loop.

Reply:
> Does this work? `a.Flush` calls `EmitBatch`, `EmitBatch` calls this `splitAndFlush`. This looks like an infinite loop.

There are two different`EmitBatch` methods. `EmitBatch`of `agentClientUDP` calls `splitAndFlush` to do the splitting work; `a.Flush` calls the jaeger thrift client stub function `EmitBatch `.